SYCL/Plugin/interop-cuda-experimental.cpp

```diff
@@ -0,0 +1,82 @@
+// REQUIRES: cuda
+
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple -lcuda %s -o %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+
+#define SYCL_EXT_ONEAPI_BACKEND_CUDA_EXPERIMENTAL 1
+#include <sycl/ext/oneapi/experimental/backend/cuda.hpp>
+#include <sycl/sycl.hpp>
+
+#include <cuda.h>
+
+#include <assert.h>
+
+void cuda_check(CUresult error) { assert(error == CUDA_SUCCESS); }
+
+template <typename refT, typename T> void check_type(T var) {
+  bool is_same = std::is_same_v<decltype(var), refT>;
+  assert(is_same);
```

Comment:
Is there a reason to not make this a `static_assert`? I suppose it would help show other compilation problems, but I think in general if there is a regression here it would be better to have us know during compilation through a static assert.

Reply:
Thanks for the suggestion. I have updated this to use a `static_assert`.